build-system/tasks/check-links.js

```diff
@@ -109,6 +109,10 @@ function checkLinksInFile(file) {
       {pattern: /localhost/},
       // codepen returns a 503 for these link checks
       {pattern: /https:\/\/codepen.*/},
+      // developers.google.com links are assumed to exist
+      {pattern: /https:\/\/developers.google.com\/.*/},
+      // anchor links should be ignored
+      {pattern: /^#/},
```

Comment:
@estherkim would you mind taking a look at this. i am not sure if this is necessary? seeing 404's on this. not sure if this is new

<img width="866" alt="Screenshot 2023-06-27 at 10 01 55 AM" src="https://github.com/ampproject/amphtml/assets/354746/9c3ad96f-60a0-4439-aa03-85e8f9484cfa">